driver-core/src/main/com/mongodb/client/model/expressions/DocumentExpression.java

```diff
@@ -25,14 +25,14 @@
 import static com.mongodb.client.model.expressions.Expressions.ofMap;
 import static com.mongodb.client.model.expressions.MqlUnchecked.Unchecked.PRESENT;
 import static com.mongodb.client.model.expressions.MqlUnchecked.Unchecked.TYPE;
+import static com.mongodb.client.model.expressions.MqlUnchecked.Unchecked.TYPE_ARGUMENT;
 
 /**
  * Expresses a document value. A document is an ordered set of fields, where the
  * key is a string value, mapping to a value of any other expression type.
  */
```

Comment:
1. `key is a string value` - we don't have keys in documents, we have fields (a pair of a name and a value). We could just "field name is a string" without saying "string value" to preserve the word "value" fully for when we talk about the field value.
2. Similarly to the documentation of a map, we should say that "no field name is repeated".